gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/GeoPoint.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.datastore;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import java.util.Objects;
+
+/**
+ * A Google Cloud Datastore GeoPoint (represented by latitude and longitude in degrees).
+ * This class is immutable.
+ *
+ * @see <a href="https://cloud.google.com/datastore/docs/concepts/entities">Google Cloud Datastore
+ *     Entities, Properties, and Keys</a>
+ */
+public final class GeoPoint extends Serializable<com.google.type.LatLng> {
```

Comment:
I wonder if this should be called LatLng instead since that is the term used in other Google APIs, for example [maps][0].

[0]:https://developers.google.com/android/reference/com/google/android/gms/maps/model/LatLng]